pages/api/users.ts

```diff
@@ -14,8 +14,12 @@ export default async function handler(req: NextApiRequest, res: NextApiResponse<
 			// validate usertype
 			if (!userType || !USER_TYPES.includes(userType)) return res.status(400).send('Invalid user type');
 
-			const users = await User.find({ userType });
-			return res.status(200).send(users);
+			let users = User.find({ userType });
+			if (userType == 'HACKER') {
+				users = users.populate('application');
```

Comment:
maybe slightly better to check this type before we do User.find() since then we can chain it to populate and it will only run one query so scales better

Reply:
I think this is still one query cuz we're not awaiting until the end